clash-lib/clash-lib.cabal

```diff
@@ -373,7 +373,7 @@ test-suite doctests
     build-depends:
       base,
       clash-lib,
-      doctest-parallel >= 0.2 && < 0.4,
+      doctest-parallel >= 0.2 && < 0.5,
```

Comment:
@martijnbastiaan Was there a reason this was still locked to `<0.4` searching the issues brought nothing up. Quickly looking over the changes from doctest 0.3 to 0.4 also didn't bring anything controversial for me. Doctest still seem to work.

Reply:
No reason. IMO we can also just make this upper-bound less, I'm not planning on big changes.